### PR TITLE
Refactor setup wizard

### PR DESCRIPTION
**Pull request by ben-kaufman**
Refactor the setup wizard into multiple pages and add Cypress tests

### CODE REVIEW

**Comment by k9ert** [comment]
@kdmukai just added you for review. Will have a look now as well. What do you think?

File: src/cryptoadvance/specter/server_endpoints/setup.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+rand = random.randint(0, 1e32)  # to force style refresh
+
+# Setup endpoint blueprint
+setup_endpoint = Blueprint("setup_endpoint", __name__)
+
+######################### Setup pages #######################################
+@setup_endpoint.route("/start/", methods=["GET"])
+@login_required
+def start():
+    app.specter.reset_setup("bitcoind")
```
Comment (k9ert) [inline src/cryptoadvance/specter/server_endpoints/setup.py:39]: Shouldn't we check here for the status? Maybe the setup is in full swing?

File: src/cryptoadvance/specter/server_endpoints/setup.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    app.specter.reset_setup("torbrowser")
+    return render_template("setup/start.jinja", specter=app.specter, rand=rand)
+
+
+@setup_endpoint.route("/tor/", methods=["GET"])
+@login_required
+def tor():
+    return render_template(
```
Comment (k9ert) [inline src/cryptoadvance/specter/server_endpoints/setup.py:47]:
```suggestion
def tor():
    ''' wizard: Setup Tor daemon (Skip / Setup Tor) '''
    return render_template(
```

File: src/cryptoadvance/specter/server_endpoints/setup.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        nextURL="setup_endpoint.node_type",
+        specter=app.specter,
+        rand=rand,
+    )
+
+
+@setup_endpoint.route("/node_type/", methods=["GET"])
+@login_required
+def node_type():
+    return render_template("setup/node_type.jinja", specter=app.specter, rand=rand)
```
Comment (k9ert) [inline src/cryptoadvance/specter/server_endpoints/setup.py:58]:
```suggestion
def node_type():
    ''' wizard: Would you like to setup a new Bitcoin node or connect to an existing one?
        (Connect existing node / Setup a new node )
    '''
    return render_template("setup/node_type.jinja", specter=app.specter, rand=rand)
```

File: src/cryptoadvance/specter/server_endpoints/setup.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+
+@setup_endpoint.route("/bitcoind/", methods=["GET"])
+@login_required
+def bitcoind():
+    return render_template("setup/bitcoind.jinja", specter=app.specter, rand=rand)
```
Comment (k9ert) [inline src/cryptoadvance/specter/server_endpoints/setup.py:64]:
```suggestion
def bitcoind():
    ''' wizard: Setup Bitcoin Core (Start the Setup!) '''
    return render_template("setup/bitcoind.jinja", specter=app.specter, rand=rand)
```

File: src/cryptoadvance/specter/server_endpoints/setup.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+
+@setup_endpoint.route("/bitcoind_datadir/", methods=["GET"])
+@login_required
+def bitcoind_datadir():
+    return render_template(
```
Comment (k9ert) [inline src/cryptoadvance/specter/server_endpoints/setup.py:70]:
```suggestion
def bitcoind_datadir():
    ''' wizard: Configure your node (Quicksync? , Start Syncing) '''
    return render_template(
```

File: src/cryptoadvance/specter/server_endpoints/setup.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+        "setup/bitcoind_datadir.jinja", specter=app.specter, rand=rand
+    )
+
+
+@setup_endpoint.route("/end/", methods=["GET"])
+@login_required
+def end():
+    return render_template("setup/end.jinja", specter=app.specter, rand=rand)
```
Comment (k9ert) [inline src/cryptoadvance/specter/server_endpoints/setup.py:78]:
```suggestion
def end():
    ''' wizard: Setup competed Successfully (Done)'''
    return render_template("setup/end.jinja", specter=app.specter, rand=rand)
```

File: src/cryptoadvance/specter/server_endpoints/setup.py [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+
+@setup_endpoint.route("/tor_setup/", methods=["GET"])
+@login_required
+def tor_from_settings():
+    return render_template(
```
Comment (k9ert) [inline src/cryptoadvance/specter/server_endpoints/setup.py:84]:
```suggestion
def tor_from_settings():
    ''' wizard: Setup Tor daemon (Setup Tor) '''
    return render_template(
```

File: src/cryptoadvance/specter/server_endpoints/setup.py [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+
+@setup_endpoint.route("/setup_bitcoind", methods=["POST"])
+@login_required
+def setup_bitcoind():
+    app.specter.config["internal_node"]["datadir"] = request.form.get(
+        "bitcoin_core_datadir", app.specter.config["internal_node"]["datadir"]
+    )
+    app.specter._save()
+    if os.path.exists(app.specter.config["internal_node"]["datadir"]):
+        if request.form["override_data_folder"] != "true":
```
Comment (k9ert) [inline src/cryptoadvance/specter/server_endpoints/setup.py:117]: I assumed that the folder gets completely purged BEFORE the new setup gets used. If that's not the case, someone with a broken chainstate or whatever broken state will never be able to recover without using "non-specter-tooling".

So imho we should have an else-clause which is purging that directory?!

File: src/cryptoadvance/specter/util/bitcoind_setup_tasks.py [hunk 9]
```diff
@@ -187,7 +169,7 @@ def setup_bitcoind_directory_thread(specter=None, quicksync=True, pruned=True):
             logger.info("No success connecting to Bitcoin Core")
         specter.check()
         specter.reset_setup("bitcoind")
-        app.specter.setup_status["stage"] = 5
+        specter.setup_status["waiting_bitcoind_datadir_setup"] = False
```
Comment (k9ert) [inline src/cryptoadvance/specter/util/bitcoind_setup_tasks.py:172]: These setup-states are a bit confusing. Can you document the data-structure a bit better, at best in the method-doc of the update_setup_status method?
https://github.com/cryptoadvance/specter-desktop/blob/13d3224f9b53dbf3ec1fd6d77e720d34775b26ef/src/cryptoadvance/specter/specter.py#L586-L594

BTW, is this still correct:
https://github.com/cryptoadvance/specter-desktop/blob/13d3224f9b53dbf3ec1fd6d77e720d34775b26ef/src/cryptoadvance/specter/specter.py#L187-L198

File: src/cryptoadvance/specter/server_endpoints/setup.py (hunk 8 above)
Comment (ben-kaufman) [reply src/cryptoadvance/specter/server_endpoints/setup.py:125]: Added

**Pull request merged**